nexus/tests/integration_tests/ip_pools.rs

```diff
@@ -269,6 +273,77 @@ async fn test_ip_pool_basic_crud(cptestctx: &ControlPlaneTestContext) {
         .expect("Expected to be able to delete an empty IP Pool");
 }
 
+#[nexus_test]
+async fn test_ip_pool_with_silo(cptestctx: &ControlPlaneTestContext) {
+    let client = &cptestctx.external_client;
+
+    // can create a pool with an existing silo by name
+    let params = IpPoolCreate {
+        identity: IdentityMetadataCreateParams {
+            name: String::from("p0").parse().unwrap(),
+            description: String::from(""),
+        },
+        silo: Some(NameOrId::Name(cptestctx.silo_name.clone())),
+        is_default: false,
+    };
+    let created_pool = create_pool(client, &params).await;
+    assert_eq!(created_pool.identity.name, "p0");
+    assert!(created_pool.silo_id.is_some());
+
+    let silo_id = created_pool.silo_id.unwrap();
```

Comment:
nit: Can just drop that assert since the unwrap would panic. `.expect` if you want a nice message.